Fix #22055: button size changes when button is selected

SirKuzalot opened the pull request:
Resolves: #22055

The size of buttons in the MainToolBar was being determined by the size of the displayed text within them. Since this text size changed when the button was selected (the text becomes bold, thus increasing in size), the button size changed. I have set the button size to always be determined by the Selected Font Size, so size does not change upon selecting the button. A re-centering of the text was also required.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [] I created a unit test or vtest to verify the changes I made (if applicable)

cbjeukendrup commented on src/framework/uicomponents/qml/MuseScore/UiComponents/PageTabButton.qml:
```diff
@@ -131,10 +131,16 @@ RadioDelegate {
             id: contentRow
             anchors.fill: parent
             spacing: root.spacing
+            width: textLabel.width
```
This change seems risky in combination with `anchors.fill`, plus that it seems undesired when there is also an icon besides the text. 
Instead, better set the `width` property of `textLabel`.

SirKuzalot commented:
@cbjeukendrup I have updated the code according to the advised changes. Changed the row width to be childrenRect.width, though this change did not provoke an alteration in neither the horizontal, icon-less menu, nor the vertical, icon-full one. However, for a matter of safe coding, I applied it anyway.

Regarding the vertical menu unalignment, it was something that I had overlooked, and as such have now added a ternary operator to decide wether the alignment will be central or to the left

cbjeukendrup commented on src/framework/uicomponents/qml/MuseScore/UiComponents/PageTabButton.qml:
```diff
@@ -145,10 +150,12 @@ RadioDelegate {
+                horizontalAlignment: root.isVertical ? Text.AlignHCenter : Text.AlignLeft
                 font: root.normalStateFont
                 text: root.title
+                elide: Text.ElideNone
```
One more question, why is this line necessary?

SirKuzalot replied:
Due to the existence of padding in the button itself, not having that line causes the text to be truncated when the button is selected.
The alternative was to add the padding to the calculation of the Text Component Width (as was originially done in when the bug existed), but I found that doing so caused the button to be unnecessarily large for the text within it. I found the implemented alternative more appealing.
As mentioned previously, I checked with the other uses of the pageTabButton and it appeared to make no difference, but I can alter the code to the Alternative if you are concerned with it provoking unwanted changes, specially in the MacOS environment, as I'm unable to test for it.

SirKuzalot replied:
Due to the way truncation is calculated, having the text box width be the exact width of the text (in other words, when the text has the Selected Font) cause the text to be truncated.

The two possible solutions that came about were to add 1 to the newly calculated width (which didn't seem like ideal practice in general) or to remove the truncation from the text in this situation (which, upon examining the remaining uses of the PageTabButton, did not create a problem, but I am unable to experiment on OS's other than Linux and Windows).

One additional option that I could make use of is remove the truncation only if the width of the text is exactly equal to the calculated width, but it would be an unnecessary check that didn't entirely make sense in theory.

If there is something I am missing (such as the text being truncated due to a pre-defined margin on text-boxes) please let me know, and I'll attempt to use that information to work on a solution that is equal to neither of the aforementioned

cbjeukendrup replied:
I found that `advanceWidth` produces the expected result on macOS. I pushed that solution to this PR. Could you try if that works for you too?

SirKuzalot replied:
The text isn«t being truncated, but I also believe it isn't being boldened when selected. Not sure if it's from the change or a different commit made somewhere along the way by a different contributor

cbjeukendrup replied:
> I also believe it isn't being boldened when selected

That seems to work fine on my side. I'll ask one of our testers to give this a final test before merging.

zacjansheski commented:
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
https://github.com/musescore/MuseScore/issues/22055 FIXED

cbjeukendrup commented:
@SirKuzalot One last question, what username did you use on MuseScore.org to sign the CLA?

SirKuzalot commented:
@cbjeukendrup The account is nicknamed SirKuzalot and the email submitted was the active email for this account: andre.peres.noronha@tecnico.ulisboa.pt

cbjeukendrup commented:
Thanks! Somehow it didn't seem to appear in the list before, but now I see it.
